platform/mbed_critical.h

```diff
@@ -89,6 +89,43 @@ void core_util_critical_section_exit(void);
  */
 bool core_util_in_critical_section(void);
 
+/**
+ * A lock-free, primitive atomic flag.
+ *
+ * Emulate C11's atomic_flag. The flag is initially in an indeterminate state
+ * unless explicitly initialised with CORE_UTIL_ATOMIC_FLAG_INIT.
+ */
+typedef struct core_util_atomic_flag {
+    uint8_t _flag;
```

Comment:
Query - Should atomic operations be extended to 16-bit and 32-bit flags as well?

Reply:
The flag is inherently an opaque 1-bit true/false boolean from an API point of view - the 8-bit storage is an implementation detail.

And I'm using `uint8_t` because the LDREXB function/intrinsic takes `uint8_t *`.

It would probably be okay to have a `bool` there and cast the pointer for LDREXB, but I hate casts. This way makes the code look neater, and means I'm not accidentally invoking undefined behaviour.

Only downside is that the compiler will insist on adding an extra instruction when I assign the `_flag` to the `bool` output of test-and-set - it thinks it needs to ensure all values from 1-255 come out as 1, when we know the thing can only be holding 0 or 1 in the first place.